### ISSUE TITLE
example cannot run properly after the update

### ISSUE THREAD

**Issue by ShengC** [http4s/rho#97]
the problem now is `HttpService` is no long a `PartialFunction`, but a `Kleisli`. Look at this code...

    def or(default: => Task[B]): Service[A, B] =
      service.mapK(_.getOrElseF(default))

where `default` is essentially `notfound` . This means once `HttpService` is instantiated, it returns 404 for any route that is not defined in the `PartialFunction`. Combining the two `HttpService` through `orElse`, which was quite natural before, now is impossible ! Yes you can still check if response of first `HttpService` is `notfound`, and then run the 2nd `HttpService` -- I don't even want to comment how ugly and fragile this would be .... 

Bottom line is I think `orElse` is one of the most, if not the very most, operators between two `HttpService`. We cannot afford to lose it - not to mention these implicit behavior changes that users will eventually find quite surprising.

**Comment by bryce-anderson** [comment]
#98 makes it work again, but I think it is a temporary fix. I agree, `orElse` is missed in this case and probably others as well. That however, is an issue for http4s proper.

**Comment by ShengC** [comment]
Thanks - I will raise the issue for http4s when I am in.

**Comment by bryce-anderson** [comment]
I already opened an issue; feel free to comment.

**Comment by ShengC** [comment]
@bryce-anderson Do we have a snapshot of `rho` before you upgraded the `http4s` dependency to 0.9 ? It is now built in Java 8. I would like to keep my current work point to the old version before I can fully wrap my head around all the changes that come with.

**Comment by ShengC** [comment]
Or even better - check out git before #94 , cut a release for it. Check out master again, upgrading the working version to 0.7-SNAPSHOT (or 0.6.5-SNAPSHOT). We certainly need some breathe room to digest all the changes that 0.9 http4s brings to rho.

**Comment by bryce-anderson** [comment]
I agree. I'll cut a release against http4s-0.8.4 but probably not against
0.7.x.

**Comment by ShengC** [comment]
Thanks - I used to compile my code against rho 0.6.0-SNAPSHOT and http4s 0.8.5-SNAPSHOT, so I'll assume 0.8.4 was built in Java 7 ?

**Comment by bryce-anderson** [comment]
Yes, 0.8.4 should be built against java-7.

**Comment by bryce-anderson** [comment]
@ShengC, rho-0.6.0 should be in central soon.

**Comment by ShengC** [comment]
Thank you - I already see it in sonatype. Will give a try .

**Comment by ShengC** [comment]
@bryce-anderson it works now. Just so you know,

    +  lazy val http4sVersion = "0.8.4"
       lazy val http4sServerVersion = if (!http4sVersion.endsWith("SNAPSHOT")) (http4sVersion.dropRight(1) + "0")

so the 0.6.0 release depends on 0.8.0 instead of 0.8.4

**Comment by bryce-anderson** [comment]
@ShengC, indeed, I forgot about that.

**Comment by bryce-anderson** [comment]
So I'm going to close this: the examples work again, and the ugly of the interface is a http4s proper issue.
